File: server/src/main/java/dev/slimevr/posestreamer/BVHSettings.java
package dev.slimevr.posestreamer;

public class BVHSettings {
	private float offsetScale = 100f;
	private float positionScale = 100f;
	private boolean writeEndNodes = false;

	public static final BVHSettings DEFAULT = new BVHSettings();
	public static final BVHSettings BLENDER = new BVHSettings(DEFAULT)
		.setOffsetScale(1f)
		.setPositionScale(1f);

	public BVHSettings() {
	}

	public BVHSettings(BVHSettings source) {
		this.offsetScale = source.offsetScale;
		this.positionScale = source.positionScale;
		this.writeEndNodes = source.writeEndNodes;
	}

	public float getOffsetScale() {
		return offsetScale;
	}

	public BVHSettings setOffsetScale(float offsetScale) {
		this.offsetScale = offsetScale;
		return this;
	}

	public float getPositionScale() {
		return positionScale;
	}

	public BVHSettings setPositionScale(float positionScale) {
		this.positionScale = positionScale;
		return this;
	}

	public boolean shouldWriteEndNodes() {
		return writeEndNodes;
	}

	public BVHSettings setWriteEndNodes(boolean writeEndNodes) {
		this.writeEndNodes = writeEndNodes;
		return this;
	}
}

File: server/core/src/main/java/dev/slimevr/posestreamer/BVHSettings.kt
package dev.slimevr.posestreamer

import dev.slimevr.tracking.processor.BoneType

class BVHSettings {
	var offsetScale: Float = 100f
		private set
	var positionScale: Float = 100f
		private set
	var rootBone: BoneType = BoneType.HIP
		private set

	constructor()

	constructor(source: BVHSettings) {
		this.offsetScale = source.offsetScale
		this.positionScale = source.positionScale
	}

	fun setOffsetScale(offsetScale: Float): BVHSettings {
		this.offsetScale = offsetScale
		return this
	}

	fun setPositionScale(positionScale: Float): BVHSettings {
		this.positionScale = positionScale
		return this
	}

	fun setRootBone(rootBone: BoneType): BVHSettings {
		this.rootBone = rootBone
		return this
	}

	companion object {
		val DEFAULT: BVHSettings = BVHSettings()
		val BLENDER: BVHSettings = BVHSettings(DEFAULT)
			.setOffsetScale(1f)
			.setPositionScale(1f)
	}
}
